client/blocks/import/capture/test/index.tsx

```diff
@@ -85,4 +85,149 @@ describe( 'CaptureInput', () => {
 
 		expect( screen.getByRole( 'button', { name: /XYZ/ } ) ).toBeInTheDocument();
 	} );
+
+	it( 'should show no TLD error', async () => {
+		const onInputEnter = jest.fn();
+		const { getByText } = render(
+			<MemoryRouter>
+				<CaptureInput onInputEnter={ onInputEnter } />
+			</MemoryRouter>
+		);
+
+		await userEvent.type( screen.getByLabelText( /Enter the URL of the site/ ), 'myblog' );
+
+		await userEvent.click( screen.getByRole( 'button', { name: /Continue/ } ) );
+
+		expect(
+			getByText(
+				'Your URL is missing a top-level domain (e.g., .com, .net, etc.). Example URL: example.com'
+			)
+		).toBeInTheDocument();
+	} );
+
+	it( 'should show no TLD error for URL with protocol', async () => {
+		const onInputEnter = jest.fn();
+		const { getByText } = render(
+			<MemoryRouter>
+				<CaptureInput onInputEnter={ onInputEnter } />
+			</MemoryRouter>
+		);
+
+		await userEvent.type( screen.getByLabelText( /Enter the URL of the site/ ), 'https://myblog' );
+
+		await userEvent.click( screen.getByRole( 'button', { name: /Continue/ } ) );
+
+		expect(
+			getByText(
+				'Your URL is missing a top-level domain (e.g., .com, .net, etc.). Example URL: example.com'
+			)
+		).toBeInTheDocument();
+	} );
+} );
+
+// Helper function to enter URL and click CTA.
+const enterUrlAndContinue = async ( url ) => {
+	await userEvent.type( screen.getByLabelText( /Enter the URL of the site/ ), url );
+	await userEvent.click( screen.getByRole( 'button', { name: /Continue/ } ) );
+};
+
+describe( 'URL Validation', () => {
+	beforeEach( () => {
+		const onInputEnter = jest.fn();
+		render(
+			<MemoryRouter>
+				<CaptureInput onInputEnter={ onInputEnter } />
+			</MemoryRouter>
+		);
+	} );
+
+	test( 'should show error for missing top-level domain', async () => {
+		await enterUrlAndContinue( 'myblog' );
+
+		expect(
+			screen.getByText(
+				'Your URL is missing a top-level domain (e.g., .com, .net, etc.). Example URL: example.com'
+			)
+		).toBeInTheDocument();
+	} );
+
+	test( 'should show error for missing top-level domain event when protocol is present', async () => {
+		await enterUrlAndContinue( 'https://myblog' );
```

Comment:
You can use dinamic to test all scenarios:


Example:
```
it.each([
  {scenario: 'A', input: 'XXX',  error: 'YYY'},
  {scenario: 'B', input: 'BB',  error: 'ZZZ'},
])(' show error ', ({scenario,  input, expected}) => {
    await enterUrlAndContinue( input );
expect(
			screen.getByText( error )
		).toBeInTheDocument();
});
```

Docs here https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
Very good suggestion, thank you! Done here https://github.com/Automattic/wp-calypso/pull/91748/commits/3447cd32549e85efbeeba51ca3f8ec7882a64e3d